integration-tests/src/test/java/tachyon/client/keyvalue/KeyValueStoreIntegrationTest.java

```diff
@@ -118,4 +121,51 @@ public void createAndOpenStoreWithMultiKeysTest() throws Exception {
     Assert.assertNull(mReader.get(KEY2));
     mReader.close();
   }
+
+  /**
+   * Tests that an iterator for an empty store has no next elements.
+   *
+   * @throws Exception when any exception happens
+   */
+  @Test
+  public void emptyStoreIteratorTest() throws Exception {
+    mWriter = sKVStores.create(mStoreUri);
+    mWriter.close();
+
+    mReader = sKVStores.open(mStoreUri);
+    KeyValueIterator iterator = mReader.iterator();
+    Assert.assertFalse(iterator.hasNext());
+  }
+
+  /**
+   * Tests that an iterator can correctly iterate over a store.
+   *
+   * @throws Exception when any exception happens
+   */
+  @Test
+  public void iteratorTest() throws Exception {
+    mWriter = sKVStores.create(mStoreUri);
```

Comment:
same argument here: two keyvalue pairs seems too few to test iterator. and we should make the test independent from implementation detail (so when we improve / change implementation, test still works)